torch/csrc/Stream.cpp

```diff
@@ -107,7 +107,7 @@ PyTypeObject THPStreamType = {
 
 void THPStream_init(PyObject* module) {
   THPStreamClass = &THPStreamType;
-  Py_TYPE(&THPStreamType) = &PyType_Type;
+  Py_SET_TYPE(&THPStreamType, &PyType_Type);
```

Comment:
[Py_SET_TYPE](https://docs.python.org/3/c-api/structures.html#c.Py_SET_TYPE) is documented for python 3.9 and up. 
```suggestion
#if PY_VERSION_HEX < 03090000
  Py_TYPE(&THPStreamType) = &PyType_Type;
#else
  Py_SET_TYPE(&THPStreamType, &PyType_Type);
#endif
```

Reply:
or is `torch/csrc/utils/python_compat.h` already included here somehow?

Reply:
`python_compat.h` is already included through `THP.h` -> `csrc/utils.h` indeed